app/src/main/java/com/example/atheneum/activities/ShowRequestInfoActivity.java

```diff
@@ -125,5 +140,92 @@ public void onClick(View v) {
             }
         });
 
+        if (!rStaus.equals("ACCEPTED")){
+            Log.i(TAG, "scan button not visible");
+            Log.i(TAG, rStaus);
+            scanBook.setVisibility(View.INVISIBLE);
+        }
```

Comment:
You should use View.Gone not View.Invisible since View.Gone will completely remove it from the layout. With View.Invisible you can still potentially click where the button should be and then you would still be able to trigger a scan.

Reply:
If you need examples of this, look at Alex's PR #189.